modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/UpNextAdapter.kt

```diff
@@ -166,9 +171,32 @@ class UpNextAdapter(
                     root.resources.getQuantityString(LR.plurals.player_up_next_header_title, header.episodeCount, header.episodeCount, time)
                 }
 
+                shuffle.isVisible = FeatureFlag.isEnabled(Feature.UP_NEXT_SHUFFLE)
+                shuffle.setShuffleDisabledButton(context)
+
+                shuffle.setOnClickListener {
+                    shuffle.setShuffleEnabledButton(context)
+                }
+
                 root.layoutParams.height = ViewGroup.LayoutParams.WRAP_CONTENT
             }
         }
+
+        private fun ImageButton.setShuffleDisabledButton(
+            context: Context,
+        ) {
```

Comment:
```suggestion
        private fun ImageButton.setShuffleDisabledButton() {
```
I don't believe you need to pass the context to this as the ImageButton extends from View which contains the member variable `context`.